data/service/serialized-data-service.mod/teach/README.md

```diff
@@ -1,76 +1,230 @@
+    },
+    "AdminRole": {
+        "prototype": "data/party/role.mjson",
+        "values": {
+            "uuid": "019a245c-1467-73c0-9c0e-c6e76846030f",
+            "name": "Administrator",
+            "permissions": ["create_users", "delete_content", "manage_settings"]
         }
     }
+}
+```
 
-However, if SerializedDataService runs in the workel while the client is in a browser, there won't be a stream to find and SerializedDataService needs to return data with a read completed operation, converting data objects to raw data, anc caching that result as seen in the rough example bellow:
+**3. User Role:**
 
-    _mapDataObjectsToRawDataPromises(dataObjects, rawDataPromises) {
-        for (let aDataObject of dataObjects) {
-            let mapping = this.mappingForObjectDescriptor(aDataObject.objectDescriptor);
-            rawDataPromises.push(mapping.mapObjectToRawData(aDataObject, {})
-            .then((rawData) => {
-                let primaryKey = this.primaryKeyForTypeRawData(aDataObject.objectDescriptor, rawData);
-                this._rawDataByObjectDescriptor(aDataObject.objectDescriptor).set(primaryKey, rawData);
-            }));
+```json
+// File: data/instance/party/roles/user.mjson
+{
+    "root": {
+        "value": { "@": "UserRole" }
+    },
+    "UserRole": {
+        "prototype": "data/party/role.mjson",
+        "values": {
+            "uuid": "019a245d-5b8a-79a0-8a12-b7c86a5d041e",
+            "name": "Standard User",
+            "permissions": ["create_content", "edit_own_content"]
         }
     }
+}
+```
+
+**Pros:**
+
+-   Efficiently load a single object (e.g., fetching one TimeZone for a user).
+-   Clear, explicit file-based referencing.
+
+**Cons:**
+
+-   High file-count, which can be difficult to manage.
+-   Requires a two-step lookup (index, then file).
+
+## 4. Single File per Type request flow
+
+### Scenario
+
+A client requests a complete list of `Country` objects (for example, to populate a dropdown). To fulfill this, it triggers a `readOperation` targeting the `Country` type.
+
+#### Steps
+
+##### 1. Request received
+
+A client executes a read operation:
+
+```js
+import { montageObject as Country } from "mod/data/model/country.mjson";
+
+async function _loadCountries() {
+    const mainService = this.application.mainService;
+    this.countries = await mainService.fetchData(Country);
+}
+```
+
+This ultimately triggers `SerializedDataService.handleReadOperation(readOperation)` method.
+The `readOperation.target` will be the `Country` object/type.
+
+##### 2. Service Validates Request
+
+```js
+if (!this.handlesType(readOperation.target)) {
+    return;
+}
+```
+
+:question: Questions:
+
+1. Does the service check whether it supports a given type? If so, how and where are supported types registered?
+
+##### 3. File Location
+
+Our service needs to know which file to open for the Country type.
+
+:question: Questions:
+
+1. How do we find the file related to a type? Should we use a map?
```

Comment:
yep, I think so too, and it needs to be part of configuring an instance of SerializedDataService as I described earlier in the comments.

Reply:
Need more info on that

Reply:
I'm not sure I understand what you're missing